sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,212 @@
+    else:        
+        satellite_number = int(goes_ts.observatory.split("-")[-1])
+        if (satellite_number <1) or (satellite_number>17):
+            raise ValueError(f"The GOES satellite number has to be between 1 and 17, {satellite_number} is given")
+
+    # check if GOES-R and whether the primary detector values are given
+    if (satellite_number>=16):  
+        if "primary_detector_a" in goes_ts.columns:
+            output = _manage_goesr_detectors(goes_ts, satellite_number, abundance=abundance)
+        else:
+            warn_user("No information about primary/secondary detectors in XRSTimeSeries, assuming primary for all")
+            output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance)
+
+    # check if the older files are passed
+    elif ("Origin" in goes_ts.meta.metas[0]) and (goes_ts.meta.metas[0]["Origin"] == "SDAC/GSFC"):
+        output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance, remove_scaling=True)
+
+    else: 
+        output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance)
+    
+    return output
+
+
+@manager.require('goes_chianti_response_table',
+                ['https://hesperia.gsfc.nasa.gov/ssw/gen/idl/synoptic/goes/goes_chianti_response_latest.fits'],
+                 '4ca9730fb039e8a04407ae0aa4d5e3e2566b93dfe549157aa7c8fc3aa1e3e04d')
+
+def _goes_chianti_temp_em(goes_ts, satellite_number, secondary=0, abundance="coronal", remove_scaling=False):
+    '''
+
+
+    Parameters
+    ----------
+    goes_ts : `~sunpy.timeseries.XRSTimeSeries`
+        The GOES XRS timeseries containing the data of both the xrsa and xrsb channels (in units of W/m**2).
+    sat: `int`
+        the GOES satellite number
+    abundance: `~str`, {`coronal` | `photospheric`}
+        Default 'coronal'. Which abundances to use for the calculation.    abundances: "coronal" or "photospheric"
+
+    Returns
+    -------
+    `~sunpy.timeseries.GenericTimeSeries`
+        Conatins the temperature and emission measure calculated from the input `goes_ts` TimeSeries.
+        The two columns are:
+            `temperature` : the temperature in MK
+            `emission measure` : the emission measure in 10^49cm^-3.
+
+    Notes
+    -----
+    url = "https://hesperia.gsfc.nasa.gov/ssw/gen/idl/synoptic/goes/goes_chianti_response_latest.fits"
+
+    Satellite number starts counting at 0. 
+    For example, for GOES 15 - you pass 14 to the response table.
+
+    '''
+
+    #--------PREP THE DATA----------#
+
+
+    longflux = goes_ts.quantity("xrsb").to(u.W/u.m**2)
+    shortflux = goes_ts.quantity("xrsa").to(u.W/u.m**2)
+
+    if "xrsb_quality" in goes_ts.columns:
+        longflux[goes_ts.to_dataframe()["xrsb_quality"]!=0] = np.nan
+        shortflux[goes_ts.to_dataframe()["xrsa_quality"]!=0] = np.nan
+
+    obsdate = parse_time(goes_ts._data.index[0])
+
+    if obsdate <= Time("1983-06-28") and satellite_number==6:
```

Comment:
Do we need a comment or something to say why this is only needed for this combination?